pkg/services/ngalert/notifier/alertmanager.go

```diff
@@ -168,13 +178,17 @@ func (am *Alertmanager) AddMigration(mg *migrator.Migrator) {
 }
 
 func (am *Alertmanager) StopAndWait() {
+	if am.stopped {
+		return
+	}
+	am.stopped = true
```

Comment:
Why the bool? I don't see a scenario where this would happen.

Reply:
Tests panicked, somewhere we are using `&Alertmanager{}` directly and thought it would be nice to have this protection anyway then

Reply:
After some chat offline, we don't need this bool. The panic was actually pointing to some bug that this PR would have added which is now fixed.